streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -53,7 +54,7 @@
 import static org.apache.kafka.common.utils.Utils.getPort;
 import static org.apache.kafka.streams.processor.internals.InternalTopicManager.WINDOW_CHANGE_LOG_ADDITIONAL_RETENTION_DEFAULT;
 
-public class StreamPartitionAssignor implements PartitionAssignor, Configurable {
+public class StreamPartitionAssignor implements PartitionAssignor, Configurable, TaskIdToPartitionsProvider {
```

Comment:
Hmm even for 1. you still need to pass in some interface so that the `StreamPartitionAssignor` can still be set back on `StreamThread/TaskManager`.
I think i'll leave it as is for now and we can refactor further in a follow up